### PR TITLE
rename remaining mentions of shamir to slip39

### PR DESCRIPTION
**Pull request by ciny**
This PR finalizes #459

### CODE REVIEW

File: core/src/apps/management/reset_device/layout.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 from trezor.ui.checklist import Checklist
 from trezor.ui.info import InfoConfirm
 from trezor.ui.scroll import Paginated
-from trezor.ui.shamir import NumInput
+from trezor.ui.slip39 import NumInput
```
Comment (tsusanka) [inline core/src/apps/management/reset_device/layout.py]: Actually, shouldn't we call it just `num_input.py`? We might use it somewhere else later?

**Comment by tsusanka** [comment]
Also, let's rename `ShamirNumInput` to `Slip39NumInput`?

File: core/src/apps/management/reset_device/layout.py (hunk 1 above)
Comment (ciny) [reply core/src/apps/management/reset_device/layout.py]: ok, I renamed it.

**Comment by tsusanka** [comment]
Please squash into a single commit and let's merge it

**Comment by ciny** [comment]
based on slack discussion I'm merging without squash since we'll be squashing when merging iss442->master.

**Pull request merged**